pkgs/by-name/fi/firefly-iota-desktop/package.nix

```
@@ -0,0 +1,40 @@
+{
+  lib,
+  fetchurl,
+  appimageTools,
+}:
+
+let
+  pname = "firefly-iota-desktop";
+  version = "2.0.12";
+  src = fetchurl {
+    url = "https://github.com/iotaledger/firefly/releases/download/desktop-iota-${version}/${pname}-${version}.AppImage";
+    sha256 = "sha256:57e45bac47195a66c7bab36199ed44b9f607b44097f77c17865809346d44e2d5";
+  };
+  appimageContents = appimageTools.extractType2 { inherit pname version src; };
+
+in
+appimageTools.wrapType2 {
+  inherit pname version src;
+
+  extraPkgs = pkgs: [ pkgs.libsecret ];
+
+  extraInstallCommands = ''
+    mkdir -p $out/share/applications $out/share/pixmaps
+    cp ${appimageContents}/desktop.desktop $out/share/applications/${pname}.desktop
+    cp ${appimageContents}/desktop.png $out/share/pixmaps/${pname}.png
+    substituteInPlace $out/share/applications/${pname}.desktop \
+      --replace-fail 'Exec=AppRun' 'Exec=${pname}' \
+      --replace-fail 'Icon=desktop' 'Icon=${pname}'
+  '';
+
+  meta = with lib; {
+    description = "Firefly IOTA Wallet";
+    homepage = "https://firefly.iota.org";
+    changelog = "https://github.com/iotaledger/firefly/releases/tag/desktop-iota-${version}";
+    license = licenses.asl20;
+    maintainers = with lib.maintainers; [ mfcrizz ];
```

using with lib is discouraged 
```suggestion
  meta = {
    description = "Firefly IOTA Wallet";
    homepage = "https://firefly.iota.org";
    changelog = "https://github.com/iotaledger/firefly/releases/tag/desktop-iota-${version}";
    license = lib.licenses.asl20;
    maintainers = with lib.maintainers; [ mfcrizz ];
```